catalog/app/components/Layout/Container.tsx

```diff
@@ -16,23 +16,15 @@ export function Provider({ children }: ProviderProps) {
   return <Ctx.Provider value={value}>{children}</Ctx.Provider>
 }
 
-export const useFullWidth = (newFullWidth?: boolean) => {
+export function useFullWidth() {
   const ctx = React.useContext(Ctx)
-  invariant(ctx, 'Context must be used within an Provider')
-  const [fullWidth, setFullWidth] = ctx
+  invariant(ctx, 'Context must be used within a Provider')
+  const [, setFullWidth] = ctx
 
   React.useEffect(() => {
-    if (typeof newFullWidth === 'undefined') return
-
-    let prev = false
-    setFullWidth((oldFullWidth) => {
-      prev = oldFullWidth
-      return newFullWidth
-    })
-    return () => setFullWidth(prev)
-  }, [setFullWidth, newFullWidth])
-
-  return fullWidth
+    setFullWidth(true)
+    return () => setFullWidth(false)
```

Comment:
if you use this in two different components, and one of them unmounts, state will be set to false -- that doesn't seem like intended behavior

Reply:
My intention was to use this hook/setter only once on the page. This is a global per-page flag, so, I don't expect it to be used differently. Nevertheless, I've added check that this hook called only once.